lib/node_modules/@stdlib/array/complex64/test/test.filter.js

```diff
@@ -0,0 +1,180 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isFunction = require( '@stdlib/assert/is-function' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+var Complex64Array = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof Complex64Array, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is an `filter` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'filter' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.filter ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.filter.call( value, predicate );
+		};
+	}
+
+	function predicate( v ) {
+		return ( realf( v ) > 0 && imagf( v ) < 0 );
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.filter( value );
+		};
+	}
+});
+
+tape( 'the method returns empty array if provided an empty complex number array', function test( t ) {
+	var arr;
+	var out;
+
+	arr = new Complex64Array();
+	out = arr.filter( predicate );
+
+	t.strictEqual( out.length, 0, 'returns expected value' );
+	t.end();
+
+	function predicate( v ) {
+		return ( realf( v ) > 0 && imagf( v ) < 0 );
+	}
+});
+
+tape( 'the method returns a new complex number array containing only those elements which satisfy a test condition', function test( t ) {
+	var expected;
+	var actual;
+	var arr;
+
+	arr = new Complex64Array( [ 1.0, -1.0, 2.0, 2.0, 3.0, -3.0 ] );
+	expected = new Complex64Array( [ 1.0, -1.0, 3.0, -3.0 ] );
+	actual = arr.filter( predicate );
+
+	t.deepEqual( actual, expected, 'returns expected value' );
```

Comment:
@Jaysukh-409 Unfortunately, I don't think this works the way you want it to. Instead, we want to reinterpret the `Complex64Array` to a `Float32Array` (see, e.g., https://github.com/stdlib-js/stdlib/blob/1464d5bac230c75ec5c98c59ad36c5e58d7d0e21/lib/node_modules/%40stdlib/array/full/test/test.js#L204) and then deep equal the reinterpreted array to an expected `Float32Array`.